skrifa/src/setting.rs

```diff
@@ -61,6 +79,15 @@ impl<T> From<([u8; 4], T)> for Setting<T> {
     }
 }
 
+impl<T: Copy> From<&([u8; 4], T)> for Setting<T> {
+    fn from(s: &([u8; 4], T)) -> Self {
+        Self {
+            selector: Tag::new_checked(&s.0[..]).unwrap_or_default(),
+            value: s.1,
+        }
+    }
+}
+
```

Comment:
looking at this, I personally think I would consider being more opinionated, and support fewer options for how a setting gets represented?

This API definitely sticks out to me (and I think to you, judging by the iteration?) as one of the current rough spots.

Just out of curiosity, how are you calling this API currently? What is your source type (your sequence of variation settings)?

Reply:
In some sense, I see the `Setting` type and its `VariationSetting` alias as being both a place to attach documentation _and_ a target for supporting various useful conversions. I think the `From` impls cost us very little but allow for a lot of flexibility for clients. For example, in the documentation I updated for the method at your request above, I note that these are all equivalent:
```rust
// slice of VariationSetting
builder.variation_settings(&[
     VariationSetting::new(Tag::new(b"wgth"), 720.0),
     VariationSetting::new(Tag::new(b"wdth"), 50.0),
])
// slice of (Tag, f32)
builder.variation_settings(&[(Tag::new(b"wght"), 720.0), (Tag::new(b"wdth"), 50.0)])
// slice of (&str, f32)
builder.variation_settings(&[("wght", 720.0), ("wdth", 50.0)])
```
I did remove the `([u8; 4], f32)` conversion as it seemed overly redundant. `(Tag, f32)` may also not be useful. I do find the `(&str, f32)` conversion particularly helpful when playing with the code which I do think is important. And fwiw, we do actually use it in vello.